Tribler/Test/Core/test_downloadconfig.py

```diff
@@ -0,0 +1,84 @@
+import os
+from tempfile import mkdtemp
+
+import shutil
+from nose.tools import raises
+
+from Tribler.Core.DownloadConfig import DownloadConfigInterface, DownloadStartupConfig, get_default_dest_dir, \
+    get_default_dscfg_filename
+from Tribler.Core.Utilities.configparser import CallbackConfigParser
+from Tribler.Core.simpledefs import DLMODE_VOD, UPLOAD, DOWNLOAD
+from Tribler.Test.Core.base_test import TriblerCoreTest
+
+
+class TestConfigParser(TriblerCoreTest):
+
+    FILE_DIR = os.path.abspath(os.path.dirname(os.path.realpath(__file__)))
+    CONFIG_FILES_DIR = os.path.abspath(os.path.join(FILE_DIR, u"data/config_files/"))
+
+    def test_downloadconfig(self):
+        dlconf = CallbackConfigParser()
+        dlconf.add_section('downloadconfig')
+        dlconf.set('downloadconfig', 'hops', 5)
+        dlcfg = DownloadConfigInterface(dlconf)
+
+        temp_dir = mkdtemp(suffix="_tribler_test_session")
+        self.assertIsInstance(dlcfg.get_dest_dir(), unicode)
+        dlcfg.set_dest_dir(temp_dir)
+        self.assertEqual(dlcfg.get_dest_dir(), temp_dir)
+
+        dlcfg.set_corrected_filename("foobar")
+        self.assertEqual(dlcfg.get_corrected_filename(), "foobar")
+
+        dlcfg.set_mode(1)
+        self.assertEqual(dlcfg.get_mode(), 1)
+
+        dlcfg.set_hops(4)
+        self.assertEqual(dlcfg.get_hops(), 4)
+
+        dlcfg.set_safe_seeding(False)
+        self.assertFalse(dlcfg.get_safe_seeding())
+
+        dlcfg.set_selected_files("foo.bar")
+        self.assertEqual(dlcfg.get_selected_files(), ["foo.bar"])
+
+        dlcfg.set_max_speed(UPLOAD, 1337)
+        dlcfg.set_max_speed(DOWNLOAD, 1338)
+        self.assertEqual(dlcfg.get_max_speed(UPLOAD), 1337)
+        self.assertEqual(dlcfg.get_max_speed(DOWNLOAD), 1338)
+
+    @raises(ValueError)
+    def test_downloadconfig_set_vod_multiple_files(self):
+        dlcfg = DownloadConfigInterface()
+        dlcfg.set_mode(DLMODE_VOD)
+        dlcfg.set_selected_files(["foo.txt", "bar.txt"])
+
+    def test_downloadconfig_copy(self):
+        dlcfg = DownloadConfigInterface()
+        dlcfg_copy = dlcfg.copy()
+
+        self.assertEqual(dlcfg_copy.get_hops(), 0)
+
+    def test_downloadstartupconfig_copy(self):
+        dlcfg = DownloadStartupConfig()
+        dlcfg_copy = dlcfg.copy()
+
+        self.assertEqual(dlcfg_copy.get_hops(), 0)
+
+    def test_startup_download_save_load(self):
+        temp_dir = mkdtemp(suffix="_tribler_test_session")
```

Comment:
Avoid creating temporary files in the tests, if they fail they will never get cleaned up. Instead create new dirs inside of `self.session_base_dir` which is already created and cleaned up by the base classes even if the test fails.